src/js/TuningSliders.js

```diff
@@ -181,33 +185,39 @@ TuningSliders.resetPidSliders = function() {
 };
 
 TuningSliders.resetGyroFilterSlider = function() {
-    if (!this.cachedGyroSliderValues) {
-        this.sliderGyroFilterMultiplier = 1;
-    }
-
     if (semver.lte(FC.CONFIG.apiVersion, API_VERSION_1_43)) {
+        if (!TuningSliders.cachedGyroSliderValues) {
+            TuningSliders.sliderGyroFilterMultiplier = 1;
+        }
         $('#sliderGyroFilterMultiplier').val(this.downscaleSliderValue(this.sliderGyroFilterMultiplier));
     } else {
         this.initGyroFilterSliderPosition();
     }
 
     this.calculateNewGyroFilters();
-    this.updateFilterSlidersDisplay();
+
+    const delayInMilliseconds = 300; //0.3 seconds
+    setTimeout(function() {
+        TuningSliders.updateFilterSlidersDisplay();
+    }, delayInMilliseconds);
 };
 
 TuningSliders.resetDTermFilterSlider = function() {
-    if (!this.cachedDTermSliderValues) {
-        this.sliderDTermFilterMultiplier = 1;
-    }
-
     if (semver.lte(FC.CONFIG.apiVersion, API_VERSION_1_43)) {
+        if (!TuningSliders.cachedDTermSliderValues) {
+            TuningSliders.sliderDTermFilterMultiplier = 1;
+        }
         $('#sliderDTermFilterMultiplier').val(this.downscaleSliderValue(this.sliderDTermFilterMultiplier));
     } else {
         this.initDTermFilterSliderPosition();
     }
 
     this.calculateNewDTermFilters();
-    this.updateFilterSlidersDisplay();
+
+    const delayInMilliseconds = 300; //0.3 seconds
+    setTimeout(function() {
+        TuningSliders.updateFilterSlidersDisplay();
+    }, delayInMilliseconds);
```

Comment:
Why `setTimeout`? In general it leads to possible memory leaks, E.G. you do something, and then navigate away. Is there a better signal? Like for example sending a message and waiting for promise to resolve?